src/osqp/new_interface.py

```diff
@@ -190,11 +190,10 @@ def derivative_iterative_refinement(self, rhs, max_iter, tol):
         sol = solver.solve(rhs)
 
         for k in range(max_iter):
-            delta_sol = solver.solve(rhs - M @ sol)
```

Comment:
As discussed on Slack with Rajiv, I moved the check towards the top to reduce the LOCs in C.